hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieCopyOnWriteTableInputFormat.java

```diff
@@ -254,6 +260,8 @@ private List<FileStatus> listStatusForSnapshotMode(JobConf job,
               .stream()
               .flatMap(Collection::stream)
               .map(fileSlice -> createFileStatusUnchecked(fileSlice, fileIndex, virtualKeyInfoOpt))
+              .filter(FileStatusOpt -> FileStatusOpt.isPresent())
```

Comment:
Instead of changing the API of the `createFileStatusUnchecked`, let's just move up this filtering to be preceding its invocation

Reply:
> Instead of changing the API of the `createFileStatusUnchecked`, let's just move up this filtering to be preceding its invocation

Do you mean that add an api `checkIfValidFileSlice` before createFileStatusUnchecked to filter the fileSlice like below
`
artitionedFileSlices.values()
              .stream()
              .flatMap(Collection::stream)
              .filter(fileSlice -> checkIfValidFileSlice(fileSclice))
              .map(fileSlice -> createFileStatusUnchecked(fileSlice, fileIndex, virtualKeyInfoOpt))
`
In `checkIfValidFileSlice` to check the 3 condition of fileSlice : (1) has baseFile: return ture ; (2) has logFile and hasn't baseFile: return false ; (3) hasn't file : throw exception.

Reply:
> Instead of changing the API of the `createFileStatusUnchecked`, let's just move up this filtering to be preceding its invocation

I'v updated the commit. Please review again. Thx for advice.

Reply:
> > Instead of changing the API of the `createFileStatusUnchecked`, let's just move up this filtering to be preceding its invocation
> 
> I'v updated the commit. Please review again. Thx for advice.

There is some wrong with the new code. I will fix it later.

Reply:
> > > Instead of changing the API of the `createFileStatusUnchecked`, let's just move up this filtering to be preceding its invocation
> > 
> > 
> > I'v updated the commit. Please review again. Thx for advice.
> 
> There is some wrong with the new code. I will fix it later.

done